uxarray/grid/utils.py

```diff
@@ -169,3 +169,75 @@ def _newton_raphson_solver_for_gca_constLat(init_cart,
         _iter += 1
 
     return np.append(y_new, constZ)
+
+
+def _get_cartesian_face_edge_nodes(face_nodes_ind, face_edges_ind,
+                                   edge_nodes_grid, node_x_values,
+                                   node_y_values, node_z_values):
+    """Helper function that constructs the face-edge connectivity for a Cartesian grid.
+    Parameters
+    ----------
+    face_nodes_ind :
+        The ith entry of Grid.face_node_connectivity
+    face_edges_ind :
+        The ith entry of Grid.face_edge_connectivity
```

Comment:
@hongyuchen1030 

Is there an explicit reason why we want this function to only index a single face, as opposed to constructing the array across all faces?

Reply:
Yes, because the `__is_pole_point` is taking each face in input.

If you do try to analyze the time complexity here, doing an array or one face is all O(log n), but doing it face-by-face is more neat.

At last. It's always good to keep each function modulized. You can always build your array with this single face function without any unnecessary treatment, but not vice versa.